src/plugins/workflow_designer/src/library/DocWorkers.cpp

```diff
@@ -873,6 +875,134 @@ bool MSAWriter::isStreamingSupport() const {
     return false;
 }
 
+/*************************************
+ * UgeneDBWriter
+ *************************************/
+void UgeneDBWriter::data2doc(Document *doc, const QVariantMap &data) {
+    data2document(doc, data, context);
+}
+
+void UgeneDBWriter::storeEntry(IOAdapter *io, const QVariantMap &data, int entryNum) {
+    streamingStoreEntry(format, io, data, context, entryNum);
+}
+
+void UgeneDBWriter::data2document(Document *doc, const QVariantMap &data, WorkflowContext *context) {
+    QScopedPointer<U2SequenceObject> seqObj(nullptr);
+    U2SequenceObject *dna = nullptr;
+    QString annotationName;
+
+    if (data.contains(BaseSlots::DNA_SEQUENCE_SLOT().getId())) {
+        SharedDbiDataHandler seqId = data[BaseSlots::DNA_SEQUENCE_SLOT().getId()].value<SharedDbiDataHandler>();
+        seqObj.reset(StorageUtils::getSequenceObject(context->getDataStorage(), seqId));
+        SAFE_POINT(nullptr != seqObj.data(), tr("UgeneDB writer: NULL sequence object"), );
+
+        U2OpStatusImpl os;
+        DNASequence seq = seqObj->getWholeSequence(os);
+        SAFE_POINT_OP(os, );
+
+        if (seq.getName().isEmpty()) {
+            int num = doc->findGObjectByType(GObjectTypes::SEQUENCE).size();
+            seq.setName(QString("unknown sequence %1").arg(num));
+        } else {
+            annotationName = getAnnotationName(seq.getName());
+        }
+
+        dna = qobject_cast<U2SequenceObject *>(doc->findGObjectByName(seq.getName()));
+        if (!dna && !seq.isNull()) {
+            dna = addSeqObject(doc, seq);
+        }
+    }
+
+    if (data.contains(BaseSlots::ANNOTATION_TABLE_SLOT().getId())) {
+        const QVariant &annsVar = data[BaseSlots::ANNOTATION_TABLE_SLOT().getId()];
+        const QList<SharedAnnotationData> atl = StorageUtils::getAnnotationTable(context->getDataStorage(), annsVar);
```

Comment:
Do not use "random" consts (or consts for locals in methods at all). 
Example below you write "QList<GObject *> relAnns" with no const at all. Having consts in one places and not others is inconsistent code style.